manifests/renew.pp

```diff
@@ -86,13 +90,37 @@
   ]).filter | $arg | { $arg =~ NotUndef and $arg != [] }
   $command = join($_command, ' ')
 
-  cron { 'letsencrypt-renew':
-    ensure   => $cron_ensure,
-    command  => $command,
-    user     => 'root',
-    hour     => $cron_hour,
-    minute   => $cron_minute,
-    monthday => $cron_monthday,
+  if timer_type == 'cron' {
+    cron { 'letsencrypt-renew':
+      ensure   => $cron_ensure,
+      command  => $command,
+      user     => 'root',
+      hour     => $cron_hour,
+      minute   => $cron_minute,
+      monthday => $cron_monthday,
+    }
+  } elsif $timer_type == 'systemd' {
+    ::systemd::unit_file { 'letsencrypt.timer':
+      content => template('letsencrypt/letsencrypt.timer.erb'),
+    }
+    ::systemd::unit_file { 'letsencrypt.service':
+      content => template('letsencrypt/letsencrypt.service.erb'),
+    }
+    service { 'letsencrypt.timer':
+      ensure    => running,
+      enable    => true,
+      hasstatus => true,
+      restart   => 'service letsencrypt.timer restart',
```

Comment:
With systemd, the command `service` is a old fashion method. The command `systemctl` should be preferred (probably with path).

Reply:
Hum ... but why is it necessary to specify the `restart` attribute ?